tarea2/tarea2.js

```diff
@@ -17,90 +18,161 @@ $botonRemover.onclick = function(){
     if (totalIntegrantes == 1){
         $botonRemover.className = "hidden";
     }
+    ocultar("#calculos");
     return false;
 }
 
 const $botonCalcular = document.querySelector("#boton-calcular")
 $botonCalcular.onclick = function(){
-    let $salarioIntegrantes = obtenerSalarios();
-    if ($salarioIntegrantes.length == 0){
-        alert("Todos los salarios son nulos, ingrese valores positivos para continuar");
+    removerErrores ();
+
+    let integrantes = {};
+    const $salarioIntegrantes = obtenerSalarios();
+
+    for (let i=0; i<totalIntegrantes; ++i){
+        integrantes[i] = {};
+        integrantes[i] = {
+            salario: $salarioIntegrantes[i],
+            "error-salario": validarSalario ($salarioIntegrantes[i])
+        }
+    }
+
+    let salariosValidos = [];
+    for (let i=0; i<totalIntegrantes; ++i){
+        let error = integrantes[i]["error-salario"];
+
+        if (error){
+            mostrarError (error);
+        }
+        else{
+            salariosValidos.push (integrantes[i].salario)
+        }
+    }
```

Comment:
Estás recorriendo 2 veces `totalIntegrantes`. Eso por lo general significa que podrías haber hecho todo en un solo for.

Fijate si los podés jungar.